src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java

```diff
@@ -517,6 +518,11 @@ public SecurityComponents createSecurityComponents() {
                 } catch (GraphServiceException e) {
                     if (e.getResponseCode() == NOT_FOUND) {
                         return null;
+                    } else if (e.getResponseCode() == BAD_REQUEST) {
+                        LOGGER.log(Level.WARNING, "Failed to lookup user with userid '" + userId + "'."
```

Comment:
one more change, can you add a `if (LOGGER.isLoggable(Level.FINE))` so that it doesn't get double logged when fine is enabled

Reply:
If I get the idea right then you want to log the warning message only if no "Fine" Log Recorder has been created in Jenkins. It appears that `if (LOGGER.isLoggable(Level.FINE))` is always true, even if no custom log recorders exist. Maybe because the log is always printed to the console on fine level?
Do you want the change regardless?

Reply:
yes please